hybrid/src/main/java/jetbrains/jetpad/hybrid/BaseHybridSynchronizer.java

```diff
@@ -329,7 +332,11 @@ public boolean isSupported(ContentKind<?> kind) {
           @Override
           public <T> T get(ContentKind<T> kind) {
             if (kind == TOKENS_CONTENT) {
-              return (T) Collections.unmodifiableList(tokens);
+              List<Token> result = new ArrayList<>(copiedTokens.size());
```

Comment:
I don't see any reason for copying 'copiedTokens' once again. Why don't you return just an unmodifiableList(copiedTokens)?